src/v/cloud_storage/cache_service.cc

```diff
@@ -1303,7 +1303,20 @@ ss::future<> cache::put(
         eptr = std::current_exception();
     }
 
+    // If we failed to write to the tmp file, we should delete it, maybe do an
+    // eager trim, and rethrow the exception.
     if (eptr) {
+        try {
+            co_await delete_file_and_empty_parents(
+              (dir_path / tmp_filename).native());
```

Comment:
`dir_path / tmp_filename` is re-used a lot in this function, worth making a named variable for?

Reply:
Agree. Will do.